src/H5FDs3comms.c

```diff
@@ -1304,23 +1321,25 @@ H5FD_s3comms_s3r_read(s3r_t *handle, haddr_t offset, size_t len, void *dest)
     else {
         /* authenticate request
          */
-        char authorization[512 + 1];
-        /*   512 := approximate max length...
-         *    67 <len("AWS4-HMAC-SHA256 Credential=///s3/aws4_request,"
-         *           "SignedHeaders=,Signature=")>
-         * +   8 <yyyyMMDD>
-         * +  64 <hex(sha256())>
-         * + 128 <max? len(secret_id)>
-         * +  20 <max? len(region)>
-         * + 128 <max? len(signed_headers)>
+        authorization = (char *)H5MM_malloc(512 + H5FD_ROS3_MAX_SECRET_TOK_LEN + 1);
+        /*   2048 := approximate max length...
+         *     67 <len("AWS4-HMAC-SHA256 Credential=///s3/aws4_request,"
+         *             "SignedHeaders=,Signature=")>
+         * +    8 <yyyyMMDD>
+         * +   64 <hex(sha256())>
+         * +  128 <max? len(secret_id)>
+         * +   20 <max? len(region)>
+         * +  128 <max? len(signed_headers)>
+         * + 1024 <max? len(session_token)>
          */
-        char buffer1[512 + 1]; /* -> Canonical Request -> Signature */
         char buffer2[256 + 1]; /* -> String To Sign -> Credential */
         char iso8601now[ISO8601_SIZE];
-        char signed_headers[48 + 1];
+        buffer1        = (char *)H5MM_malloc(512 + H5FD_ROS3_MAX_SECRET_TOK_LEN +
+                                             1); /* -> Canonical Request -> Signature */
+        signed_headers = (char *)H5MM_malloc(48 + H5FD_ROS3_MAX_SECRET_KEY_LEN + 1);
```

Comment:
Should add some error checking here to make sure `authorization`, `buffer1` and `signed_headers` were properly allocated before using them

Reply:
The additional error checkings have been implemented.